src/Microsoft.Health.Fhir.Shared.Api/Controllers/ValidateController.cs

```diff
@@ -0,0 +1,62 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Linq;
+using System.Threading.Tasks;
+using Hl7.Fhir.Model;
+using MediatR;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Options;
+using Microsoft.Health.Fhir.Api.Configs;
+using Microsoft.Health.Fhir.Api.Features.ActionResults;
+using Microsoft.Health.Fhir.Api.Features.Audit;
+using Microsoft.Health.Fhir.Api.Features.Filters;
+using Microsoft.Health.Fhir.Api.Features.Routing;
+using Microsoft.Health.Fhir.Api.Features.Security;
+using Microsoft.Health.Fhir.Core.Extensions;
+using Microsoft.Health.Fhir.Core.Features.Operations;
+using Microsoft.Health.Fhir.Core.Messages.Operation;
+using Microsoft.Health.Fhir.ValueSets;
+
+namespace Microsoft.Health.Fhir.Api.Controllers
+{
+    [ServiceFilter(typeof(AuditLoggingFilterAttribute))]
+    [ServiceFilter(typeof(OperationOutcomeExceptionFilterAttribute))]
+    [ServiceFilter(typeof(ValidateContentTypeFilterAttribute))]
+    [ValidateResourceTypeFilter]
+    [ValidateModelState]
+    [Authorize(PolicyNames.FhirPolicy)]
+    public class ValidateController : Controller
+    {
+        private readonly IMediator _mediator;
+        private readonly FeatureConfiguration _features;
+
+        public ValidateController(IMediator mediator, IOptions<FeatureConfiguration> features)
+        {
+            _mediator = mediator;
```

Comment:
Null checks?

Reply:
Added

Reply:
I think it will be better to do an `EnsureArg.IsNotNull(features?.Value, nameof(features))` to make sure both are not null.

Reply:
I don't know why that isn't showing here. In my local view there are EnsureArg.IsNotNulls for both of them and VS is telling me I have no uncommitted changes.

Reply:
For some reason GitHub wasn't showing all the commits by default. At the top if you select 'Changes from' and pick all commits you should see it.

Reply:
I wasn't referring to the `mediator` null check actually. I was referring to `features.` It would be good to check whether `features.Value` is also not null (apart from checking if `features` is not null) since that is the actual `FeatureConfiguration` object we use. Sorry for the confusion.

Reply:
No, my bad. I didn't see the checks and then quick read your comment. That is a good idea to check, I'll add that.